src/OrchardCore.Modules/OrchardCore.Search.Elastic/Services/ElasticIndexManager.cs

```diff
@@ -0,0 +1,228 @@
+                }
+
+                _timestamps[indexName] = _clock.UtcNow;
+            }
+            return elasticTopDocs;
+        }
+
+        private ElasticDocument CreateElasticDocument(DocumentIndex documentIndex)
+        {
+            ElasticDocument elasticDocument = new ElasticDocument(documentIndex.ContentItemId.ToLowerInvariant());
+            foreach (var entry in documentIndex.Entries)
+            {
+                if (entry.Name.Contains("Analyzed") || entry.Name.Contains("Sanitize") || entry.Name.Contains("Normalized"))
+                {
+                    continue;
+                }
+
+                switch (entry.Type)
+                {
+                    case DocumentIndex.Types.Boolean:
+                        // store "true"/"false" for booleans
+                        elasticDocument.Set(entry.Name, (bool)(entry.Value));
+                        break;
+
+                    case DocumentIndex.Types.DateTime:
+                        if (entry.Value != null)
+                        {
+                            if (entry.Value is DateTimeOffset)
+                            {
+                                elasticDocument.Set(entry.Name, ((DateTimeOffset)(entry.Value)).UtcDateTime);
+                            }
+                            else
+                            {
+                                elasticDocument.Set(entry.Name, ((DateTime)(entry.Value)).ToUniversalTime());
+                            }
+                        }
+                        //else
+                        //{
+                        //    elasticDocument.Set(entry.Name, null);
+                        //}
+                        break;
+
+                    case DocumentIndex.Types.Integer:
+                        if (entry.Value != null && Int32.TryParse(entry.Value.ToString(), out var value))
+                        {
+                            elasticDocument.Set(entry.Name, Convert.ToInt32(entry.Value));
+                        }
+                        //else
+                        //{
+                        //    elasticDocument.Set(entry.Name, null);
+                        //}
+
+                        break;
+
+                    case DocumentIndex.Types.Number:
+                        if (entry.Value != null)
+                        {
+                            elasticDocument.Set(entry.Name, Convert.ToDouble(entry.Value));
+                        }
+                        //else
+                        //{
+                        //    elasticDocument.Set(entry.Name, null);
+                        //}
+                        break;
+
+                    case DocumentIndex.Types.Text:
+                        if (entry.Value != null && !String.IsNullOrEmpty(Convert.ToString(entry.Value)))
+                        {
+                            elasticDocument.Set(entry.Name, Convert.ToString(entry.Value));
```

Comment:
This needs to also use the DocumentIndexOptions to define if the field is Analyzed or Stored.

Reply:
@Skrypt 

All text fields are analyzed by default in Elastic Search!
(https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-index-search-time.html)

Stored is really an overhead and only required if we are processing thousands of large documents.
By default Elastic will store the entire document into a field called _source and retrieves them when asked them from Index itself.
https://www.elastic.co/guide/en/elasticsearch/reference/7.12/search-fields.html

I just kept both of these for later implementations as noted in my first comment with PR. (Advanced Scenarios) ;-)

I think, we should 
1) first solid the basic functionalities, merge with Dev (More people can provide feedback)
2) Define an architecture leading to abstraction for (Indexing, Querying, Search based components) [This is mandatory so that developers can swap search providers without needed to code)
3) Build advanced features per provider on the go.

Reply:
Ok, I looked at these documentation pages and they strongly suggest not adding a Stored custom field because it is already in the _source field. This makes sense. Now, we need to see if we shouldn't do the same with the Lucene implementation so that it be consistent then. Though, from the documentation they say that they allow still to do it so if it's an option we might want to keep it still.

Reply:
@Skrypt , We must store with with Lucene as there is no way, to build a document when retrieved from Lucene.
Unfortunately, I had to work with Lucene, Solr and Elastic and Azure Search last 10 years at some point of time!

Reply:
@Skrypt for some reason, while debugging, i found that while added fields to Lucene everything was turned to "String" in OrcharcdCore.Lucene. I'm not sure if that changed, (I remember debugging OrchardCore.Lucene around 4 months back).

I might be wrong though! ;-)

Reply:
Where do you see this? Because If I'm looking at LuceneIndexManager we use Int32Field, DoubleField and StringField.

Reply:
@Skrypt , I see your point! I have observed though if field has no data, it is set to null and document will not have this field at all!